Gemfile

```diff
@@ -0,0 +1,13 @@
+source "https://rubygems.org"
+
+ruby ">=2.3"
+
+gem "asciidoctor", "1.5.8"           # Used by the docs build
+gem "thread_safe", "0.3.6"           # Used by asciidoctor
+gem "asciidoctor-diagram", "1.5.12"  # Speculative
+gem "asciimath", "1.0.8"             # Speculative
+
+group :test do
+  gem "rspec", "3.8.0"
+  gem "rubocop", "0.64.0"
+end
```

Comment:
The same comment holds true here: typically, `Gemfile`s have more lenient version specifications. Unless you need to pin to the very specific versions, you might want to add some leniency with the twiddle waka (`~`) so that you'll install the latest versions. That is, if you trust the gem maintainers to follow semver ; )

Reply:
I want the docker image to be as consistent as possible. I'm ok with using the `~` so long as I can use the lockfile to keep it consistent though.

Reply:
I think this approach is good now. Although you are committing the `Gemfile.lock`, if you ever want to update all gems at once, you can just run bundle install and commit the `Gemfile.lock` again. Otherwise, you would have to keep your eye on each gem individually to know when updates are available and edit this file.

Reply:
> Otherwise, you would have to keep your eye on each gem individually to know when updates are available and edit this file.

Right! I think we'll want to periodically blast the lockfile and `bundle install` again to see what we get.